### PR TITLE
Remove usage of @test with exceptions in plugin tests

### PR DESCRIPTION
**Pull request by Damans227**
Linked to `@Test(expectedExceptions...)` cleanup issue #6974

### CODE REVIEW

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 1]
```diff
@@ -337,13 +338,15 @@ public void testReadNotFound()
             fs.initialize(new URI("s3n://test-bucket/"), new Configuration(false));
             fs.setS3Client(s3);
             try (FSDataInputStream inputStream = fs.open(new Path("s3n://test-bucket/test"))) {
-                inputStream.read();
+                assertThatThrownBy(() -> inputStream.read())
+                        .isInstanceOf(IOException.class)
+                        .hasMessageMatching(".*Failing getObject call with " + HTTP_NOT_FOUND + ".*");
```
Comment (alexjo2144) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:343]: You could do `hasMessageContaining` and drop the `.*` on both ends

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 2]
```diff
@@ -395,7 +400,9 @@ public void testCreateWithStagingDirectoryFile()
             conf.set(S3_STAGING_DIRECTORY, staging.toString());
             fs.initialize(new URI("s3n://test-bucket/"), conf);
             fs.setS3Client(s3);
-            fs.create(new Path("s3n://test-bucket/test"));
+            assertThatThrownBy(() -> fs.create(new Path("s3n://test-bucket/test")))
+                    .isInstanceOf(IOException.class)
+                    .hasMessageMatching("Configured staging path is not a directory: .*");
```
Comment (alexjo2144) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:405]: Similarly, `hasMessageStartingWith`?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java (hunk 1 above)
Comment (Damans227) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:343]: Thanks for the feedback. I will go ahead and make these changes.

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 3]
```diff
@@ -353,7 +356,9 @@ public void testReadForbidden()
             fs.initialize(new URI("s3n://test-bucket/"), new Configuration(false));
             fs.setS3Client(s3);
             try (FSDataInputStream inputStream = fs.open(new Path("s3n://test-bucket/test"))) {
-                inputStream.read();
+                assertThatThrownBy(() -> inputStream.read())
+                        .isInstanceOf(IOException.class)
+                        .hasMessageMatching(".*Failing getObject call with " + HTTP_FORBIDDEN + ".*");
```
Comment (alexjo2144) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:361]: `hasMessageContaining`

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 4]
```diff
@@ -460,7 +467,9 @@ public void testGetMetadataForbidden()
             s3.setGetObjectMetadataHttpCode(HTTP_FORBIDDEN);
             fs.initialize(new URI("s3n://test-bucket/"), new Configuration(false));
             fs.setS3Client(s3);
-            fs.getS3ObjectMetadata(new Path("s3n://test-bucket/test"));
+            assertThatThrownBy(() -> fs.getS3ObjectMetadata(new Path("s3n://test-bucket/test")))
+                    .isInstanceOf(IOException.class)
+                    .hasMessageMatching(".*Failing getObjectMetadata call with " + HTTP_FORBIDDEN + ".*");
```
Comment (alexjo2144) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:472]: `hasMessageContaining`

**Review by alexjo2144 — COMMENTED** [review]
Sorry for the delayed re-review. There were a few more of these nit-picks do you mind updating them as well?

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java (hunk 4 above)
Comment (Damans227) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java]: Done!

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java (hunk 3 above)
Comment (Damans227) [reply plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java]: Done!

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 5]
```diff
@@ -353,7 +356,9 @@ public void testReadForbidden()
             fs.initialize(new URI("s3n://test-bucket/"), new Configuration(false));
             fs.setS3Client(s3);
             try (FSDataInputStream inputStream = fs.open(new Path("s3n://test-bucket/test"))) {
-                inputStream.read();
+                assertThatThrownBy(() -> inputStream.read())
+                        .isInstanceOf(IOException.class)
+                        .hasMessageContaining(".*Failing getObject call with " + HTTP_FORBIDDEN + ".*");
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:361]: The expected message is wrong. 
```suggestion
                assertThatThrownBy(inputStream::read)
                        .isInstanceOf(IOException.class)
                        .hasMessageContaining("Failing getObject call with " + HTTP_FORBIDDEN);
```

File: plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java [hunk 6]
```diff
@@ -460,7 +467,9 @@ public void testGetMetadataForbidden()
             s3.setGetObjectMetadataHttpCode(HTTP_FORBIDDEN);
             fs.initialize(new URI("s3n://test-bucket/"), new Configuration(false));
             fs.setS3Client(s3);
-            fs.getS3ObjectMetadata(new Path("s3n://test-bucket/test"));
+            assertThatThrownBy(() -> fs.getS3ObjectMetadata(new Path("s3n://test-bucket/test")))
+                    .isInstanceOf(IOException.class)
+                    .hasMessageContaining(".*Failing getObjectMetadata call with " + HTTP_FORBIDDEN + ".*");
```
Comment (ebyhr) [inline plugin/trino-hive/src/test/java/io/trino/plugin/hive/s3/TestTrinoS3FileSystem.java:472]:
```suggestion
                    .hasMessageContaining("Failing getObjectMetadata call with " + HTTP_FORBIDDEN);
```

**Review by ebyhr — APPROVED** [review]

**Comment by ebyhr** [comment]
Merged, thanks!

**Pull request merged**